### PR TITLE
Auto-discover caBundle when missing (take 2)

### PR DESCRIPTION
**Pull request by JacobGabrielson**
**Issue, if available:**

This is a prerequisite for https://github.com/awslabs/karpenter/issues/626 but does not fix it.

**Description of changes:**

fix problems with https://github.com/awslabs/karpenter/pull/636/files

1. conflict between knative and controllerruntime was causing webhook to crash, and, 
2. CAData was empty so launch template created was invalid

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/apis/provisioning/v1alpha3/provisioner_defaults.go [hunk 1]
```diff
@@ -17,65 +17,35 @@ package v1alpha3
...
-		if errors.Is(err, os.ErrNotExist) {
-			return nil, nil
-		}
+		logging.FromContext(ctx).Infof("Unable to discover caBundle, loading transport config, %v", err)
 		return nil, err
 	}
-	encoded := base64.StdEncoding.EncodeToString(binary)
-	return &encoded, nil
+	_, err = transport.TLSConfigFor(transportConfig) // fills in CAData!
+	logging.FromContext(ctx).Infof("Discovered caBundle, length %d", len(transportConfig.TLS.CAData))
```
Comment (ellistarn) [inline pkg/apis/provisioning/v1alpha3/provisioner_defaults.go:49]: This should minimally be a debug statement like the other discovery bits. How often does this log line print?

File: pkg/utils/restconfig/restconfig.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import (
+	"context"
+
+	"k8s.io/client-go/rest"
+)
+
+type contextKey string
+
+const (
+	restKey contextKey = "restKey"
```
Comment (ellistarn) [inline pkg/utils/restconfig/restconfig.go:12]: consider a struct and this style: https://github.com/knative/pkg/blob/b558677ab03404ed118ba3e179a7438fe2d8509a/apis/contexts.go#L28

**Review by ellistarn — COMMENTED** [review]
Looks good! Thanks for driving this through

File: pkg/apis/provisioning/v1alpha3/provisioner_defaults.go (hunk 1 above)
Comment (JacobGabrielson) [reply pkg/apis/provisioning/v1alpha3/provisioner_defaults.go:53]: Only when creating a launch template

File: pkg/utils/restconfig/restconfig.go (hunk 2 above)
Comment (JacobGabrielson) [reply pkg/utils/restconfig/restconfig.go:12]: Ok

File: pkg/apis/provisioning/v1alpha3/provisioner_defaults.go (hunk 1 above)
Comment (JacobGabrielson) [reply pkg/apis/provisioning/v1alpha3/provisioner_defaults.go]: done

File: pkg/utils/restconfig/restconfig.go (hunk 2 above)
Comment (JacobGabrielson) [reply pkg/utils/restconfig/restconfig.go]: done

**Review by ellistarn — DISMISSED** [review]

**Review by ellistarn — APPROVED** [review]

**Pull request merged**